### PR TITLE
[nnc] Added NNC lowerings for t/transpose/permute/expand + other cleaning

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56718 [WIP] Added skeleton of python dispatch key
* **#57426 [nnc] Added NNC lowerings for t/transpose/permute/expand + other cleaning**

Differential Revision: [D28293191](https://our.internmc.facebook.com/intern/diff/D28293191)

### CODE REVIEW

File: torch/lib/pyston3.8/site-packages/caffe2/contrib/aten/aten_test.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
+
+
+
+
+
```
Comment (ZolotukhinM) [inline torch/lib/pyston3.8/site-packages/caffe2/contrib/aten/aten_test.py:5]: Ouch, rebase went wrong?

File: torch/csrc/jit/tensorexpr/kernel.cpp [hunk 2]
```diff
@@ -2426,9 +2423,59 @@ Tensor* tensorexpr::computeOperandValue(
...
+      if (shape.size() == 1) {
+        return new Tensor(c10::get<BufHandle>(inputs[0]).node(), nullptr);
+      }
+      return computeOperandValue(
+          aten::transpose,
+          {inputs[0], (int64_t)1, (int64_t)0},
+          outputShape,
+          outputType);
+    }
+    case aten::transpose: {
```
Comment (ZolotukhinM) [inline torch/csrc/jit/tensorexpr/kernel.cpp:2440]: Can we add some correctness tests for these? We could write them without relying on the fuser, we can feed graphs directly to `TensorExprKernel` like it is done, e.g. here:
https://github.com/pytorch/pytorch/blob/b5b158a6c6de94dfb983b447fa33fea062358844/test/test_tensorexpr_pybind.py#L125-L134

**Review by ZolotukhinM — COMMENTED** [review]
Generally looks good. I'd prefer to have some tests with it and also it seems that some unrelated stuff has creeped in

File: test/test_tensorexpr_pybind.py [hunk 3]
```diff
@@ -206,5 +206,104 @@ def forward(self, x, y):
...
+  %3 : Float(4, 3, strides=[4, 1], requires_grad=0, device=cpu) = aten::t(%a.1)
+  return (%3)
+        """
+        graph = torch._C.parse_ir(graph_str)
+
+        kernel = torch._C._te.TensorExprKernel(graph)
+        res1 = kernel.run((x,))
+        res2 = kernel.fallback((x,))
+        correct = f(x)
+        np.testing.assert_allclose(res1.numpy(), correct.numpy(), atol=2e-3)
```
Comment (ZolotukhinM) [inline test/test_tensorexpr_pybind.py:228]: Nit: I know we use this `2e-3` accuracy in some existing tests, but I think we really should be more accurate than this :) Can we change it to `1e-5`?

File: torch/csrc/jit/tensorexpr/external_functions.cpp [hunk 4]
```diff
@@ -225,6 +225,24 @@ void nnc_aten_addmm(
   }
 }
 
+void nnc_aten_digamma(
```
Comment (ZolotukhinM) [inline torch/csrc/jit/tensorexpr/external_functions.cpp:228]: It would be good to add a test for this too, like we have for other external calls (in https://github.com/pytorch/pytorch/blob/master/test/cpp/tensorexpr/test_external_calls.cpp)

File: torch/csrc/jit/tensorexpr/kernel.cpp [hunk 5]
```diff
@@ -3223,4 +3276,4 @@ void TensorExprKernel::runFast(
 
   // Call the kernel.
   codegen_->call_raw(args);
-}
+}
```
Comment (ZolotukhinM) [inline torch/csrc/jit/tensorexpr/kernel.cpp:3279]: Nit: unnecessary whitespace edit.

**Review by ZolotukhinM — APPROVED** [review]
Looks good!

**Comment by Chillee** [comment]
@chillee has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28293191).
